src/matlab2tikz.m

```diff
@@ -4910,7 +4918,30 @@ function warnFacetedInterp(m2t)
 %
 % Example of usage:
 %              join(m2t, cellstr, ',')
-    newstr = m2tstrjoin(cellstr, delimiter, m2t.ff);
+    if isempty(cellstr)
+        newstr = '';
+        return
+    end
+
+    % convert all values to strings first
+    nElem = numel(cellstr);
+    for k = 1:nElem
+        if isnumeric(cellstr{k})
+            cellstr{k} = sprintf(m2t.ff, cellstr{k});
+        elseif iscell(cellstr{k})
+            cellstr{k} = join(m2t, cellstr{k}, delimiter);
+            % this will fail for heavily nested cells
+        elseif ~ischar(cellstr{k})
+            error('matlab2tikz:join:NotCellstrOrNumeric',...
+                'Expected cellstr or numeric.');
+        end
+    end
+
+    % inspired by strjoin of recent versions of MATLAB
+    newstr = cell(2,nElem);
+    newstr(1,:)         = reshape(cellstr, 1, nElem);
+    newstr(2,1:nElem-1) = {delimiter}; % put delimiters in-between the elements
+    newstr = [newstr{:}];
```

Comment:
No problem, here for this :). Even with smartgit it took me a while to figure out how to solve conflicts, and can't say I am always 100% confident and end up double-checking the code side by side...